tensorflow_addons/utils/ensure_tf_install.py

```diff
@@ -24,7 +24,7 @@
 import tensorflow as tf
 
 MIN_TF_VERSION = "2.2.0"
-MAX_TF_VERSION = "2.3.0"
+MAX_TF_VERSION = "2.4.0"
```

Comment:
So does this mean that we are compatibile with 2.3.x for the upstream version policy?

Reply:
Yes it should. TF patch version should have no breaking changes. I'm open to restructuring how this is written per your comment in the issue but for now this is how it was previously set up and I'd like to fix our broken package

Reply:
If we can trust upstream versioning contract without our CI testing It Is ok for me

Reply:
I think it should be ok for this.

Reply:
Ok but you need to be aware that `tf.experimental` is not in the contract for minor versions and we have already some experimental call in TFA (we could add other dependecies on experimental in the future):

See https://www.tensorflow.org/guide/versions?hl=en#what_is_not_covered